Make Biocro use met workflow

infotroph opened the pull request:
Probably do not pull yet -- needs tests and cleanup. @ankurdesai am I doing the met conversions right? @dlebauer should this whole laundry list go in one pull, or should I split it up?

## Description

Addresses #1283 by converting BioCro met format from CF (with files processed on the fly) to CSVs pre-generated by met2model.BIOCRO.

## Motivation and Context

BioCro is currently registered as accepting CF met files, but its handling of them is idiosyncratic and inefficient, and does not work properly in the web interface (#1283).

This PR moves all format conversion out to met2model.BIOCRO and changes the registered met input from CF met to `biocromet` (text/csv). 

@dlebauer note that this removes the `met.uncertainty` and `irrigation` arguments from run.biocro. I believe I remember you okaying that in a Skype conversation, but I neglected to make a note -- if they need to stay, help wanted deciding how to handle them.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue) <!-- please add issue number -->
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
    - [ ] Document change to run.biocro inputs
    - [ ] Update met2model.BIOCRO parameters
    - [ ] Deprecate (or delete immediately?) PEcAn.data.atmosphere::load.cfmet. Provide equivalent functionality as PEcAn.BIOCRO::load.cfmet.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
    - [ ] change existing run.biocro tests to assume external conversion
    - [ ] move existing load.cfmet tests from data.atmosphere, add more
- [x] All new and existing tests passed.
    - [x] Fix currently failing model2netcdf tests.
    - [x] Update tests/pecan64.biocro.xml to use CSV weather

infotroph commented on models/biocro/R/met2model.BIOCRO.R:
```diff
@@ -16,23 +16,60 @@
+  library(PEcAn.all)
+  dir.create(file.path(outfolder), recursive = TRUE, showWarnings = FALSE)
+  years_wanted <- lubridate::year(start_date):lubridate::year(end_date)
+  out.prefix <- "BIOCRO_MET_INPUT"
```
@ankurdesai is there an output naming convention to follow here? Currently produces files named like `/home/carya/output/dbfiles/AmerifluxLBL_BIOCRO_site_1-35//BIOCRO_MET_INPUT.2007.csv`

ankurdesai replied:
Looking at the SIPNET workflow -> 
  /home/carya/output/dbfiles/AmerifluxLBL_CF_gapfill_site_0-622/AMF_US-NR1_BASE_HH_9-1.2004.nc gets converted in met2model to
  /home/carya/output/dbfiles/AmerifluxLBL_SIPNET_site_0-772/AMF_US-NR1_BASE_HH_9-1.2004-01-01.2004-12-31.clim

In this case, the directory name keeps the formatname_modelname_siteID/, the basename passed from the file is forwarded through, and then start date and end date are appended, and then the extension is matching what the model expects (sipnet expects .clim ASCII files)

dlebauer commented:
@ankurdesai the load.cfmet and temporal.downscaling functions were designed to work on the PEcAn CF standard met files, so they aren't model specific even though only the biocro module currently uses them.

`load.cfmet` generalizes what I think is done in many of the model2netcdf.MODEL functions ... loading variables and subsetting by date. The only conversions it does is of time, to allow subsetting. The get.ncvector function also does a few things that coerce CF to PEcAn-CF (e.g. precip from kg/m2--> mm and surface_pressure --> air_pressure). This function could be a handy utility for new models or in refactoring existing code (e.g. most met2model functions have a dozen or so lines of `ncvar_get`, a conversion of "days since 1700-01-01" and a way to identify the correct lat/lon coordinates). 

`temporal.downscaling` acts on the output of `load.cfmet`, which should be fairly strictly PEcAn-CF compliant. Ideally this function would be merged or replaced by the work that you and @jsimkins2 have been doing, and now would be a good time to do it. How does the current met workflow handle downscaling? Does it take, e.g. a 3hr CF file and create one that is downscaled to one hour?

dlebauer commented on models/biocro/R/run.biocro.R:
```diff
@@ -2,56 +2,23 @@
 #' @export
 #' @author David LeBauer
-run.biocro <- function(lat, lon, metfile, soil.nc = NULL, config = config, coppice.interval = 1, 
+run.biocro <- function(lat, lon, metpath, soil.nc = NULL, config = config, coppice.interval = 1,
```
you can also remove the `met.uncertainty` and `irrigation` arguments from this. Would probably be better to either a) add a new variable in the met files like irrigation_flux that can be added to precipitation_flux explicitly inside the code or have a separate irrigation file. But that is not currently a priority. Implementation of a scheme for met (and other driver) uncertainty is discussed in #648

dlebauer commented on models/biocro/R/run.biocro.R:
```diff
@@ -142,6 +117,7 @@ run.biocro <- function(lat, lon, metfile, soil.nc = NULL, config = config, coppi
                            photoControl = l2n(config$pft$photoParms))
       
     }
+    # TODO return the whole BioGro result instead of selected columns?
```
It is easy enough to add columns here ... if you return all of the results the files can increase in size a few fold, which can get quite large (especially for ensembles and regional / hourly output). Feel free to add any additional variables you think would be of interest.

infotroph commented:
@dlebauer @ankurdesai I propose to merge this PR now-ish and push documentation and new tests separately -- this one is getting hard to track. 

Status:
- [x] addressed your review comments
- [x] existing in-package tests for PEcAn.BIOCRO and PEcAn.data.atmosphere all pass (including some that were broken before.)
- [x]  updated `./web.worflow.R --settings ./tests/pecan64.biocro.xml` to point toward a CSV-formatted weather file, added said file to test data folder.
- [x] I don't hear consensus on whether to move load.cfmet into biocro or leave it in data.atmosphere, so I'm leaving it in place for the moment. 

Propose to address in two new PRs:
* Devtools noise:
    - Recompile documentation and commit updated Rd versions
    - Fix formatting errors that are making `R CMD check` complain
    - Check that all arguments still match their descriptions
    - Finish cleaning up imports and adding explicit package:: calls
* Tests:
    - New tests covering load.cfmet, cf2biocro, run.biocro
    - Check more error conditions in existing write.configs tests
    - Test met2model.BIOCRO on more sources
